src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -582,6 +590,83 @@ public Builder addOriginatingElement(Element originatingElement) {
       return this;
     }
 
+    public Builder alwaysQualify(String... simpleNames) {
+      checkArgument(simpleNames != null, "simpleNames == null");
+      for (String name : simpleNames) {
+        checkArgument(
+            name != null,
+            "null entry in simpleNames array: %s",
+            Arrays.toString(simpleNames)
+        );
+        alwaysQualifiedNames.add(name);
+      }
+      return this;
+    }
+
+    /**
+     * Call this to always fully qualify any types that would conflict with possibly nested types of
+     * this {@code typeElement}. For example - if the following type was passed in as the
+     * typeElement:
+     *
+     * <pre><code>
+     *   class Foo {
```

Comment:
I think a better example is when a nested type clashes with a well-known Java type. I like the example presented in #77.

Reply:
Done in my added commits 👍